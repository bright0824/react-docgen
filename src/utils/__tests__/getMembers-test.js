/*
 *  Copyright (c) 2015, Facebook, Inc.
 *  All rights reserved.
 *
 *  This source code is licensed under the BSD-style license found in the
 *  LICENSE file in the root directory of this source tree. An additional grant
 *  of patent rights can be found in the PATENTS file in the same directory.
 *
 */

/*global jest, describe, beforeEach, it, expect*/

jest.disableAutomock();

describe('getMembers', () => {
  var expression;
  var getMembers;
  var memberExpressionPath;

  beforeEach(() => {
    getMembers = require('../getMembers').default;
    ({expression} = require('../../../tests/utils'));
    memberExpressionPath = expression('foo.bar(123)(456)[baz][42]');
  });

  it('finds all "members" "inside" a MemberExpression', () => {
    var members = getMembers(memberExpressionPath);

    //bar(123)
    expect(members[0].path.node.name).toEqual('bar');
    expect(members[0].computed).toBe(false);
    expect(members[0].argumentsPath.get(0).node.value).toEqual(123);
    //[baz]
    expect(members[1].path.node.name).toEqual('baz');
    expect(members[1].computed).toBe(true);
    expect(members[1].argumentsPath).toBe(null);
    //[42]
    expect(members[2].path.node.value).toEqual(42);
    expect(members[2].computed).toBe(true);
    expect(members[2].argumentsPath).toBe(null);
  });

});
