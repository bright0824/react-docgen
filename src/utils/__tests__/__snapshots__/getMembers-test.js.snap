// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMembers finds all "members" "inside" a MemberExpression 1`] = `
Array [
  Object {
    "argumentsPath": Node {
      "arguments": Array [
        Node {
          "end": 12,
          "loc": SourceLocation {
            "end": Position {
              "column": 12,
              "line": 1,
            },
            "start": Position {
              "column": 9,
              "line": 1,
            },
          },
          "raw": "123",
          "start": 9,
          "type": "Literal",
          "value": 123,
        },
      ],
      "callee": Node {
        "computed": false,
        "end": 8,
        "innerComments": undefined,
        "leadingComments": undefined,
        "loc": SourceLocation {
          "end": Position {
            "column": 8,
            "line": 1,
          },
          "start": Position {
            "column": 1,
            "line": 1,
          },
        },
        "object": Node {
          "end": 4,
          "innerComments": undefined,
          "leadingComments": undefined,
          "loc": SourceLocation {
            "end": Position {
              "column": 4,
              "line": 1,
            },
            "identifierName": "foo",
            "start": Position {
              "column": 1,
              "line": 1,
            },
          },
          "name": "foo",
          "start": 1,
          "trailingComments": undefined,
          "type": "Identifier",
        },
        "property": Node {
          "end": 8,
          "innerComments": undefined,
          "leadingComments": undefined,
          "loc": SourceLocation {
            "end": Position {
              "column": 8,
              "line": 1,
            },
            "identifierName": "bar",
            "start": Position {
              "column": 5,
              "line": 1,
            },
          },
          "name": "bar",
          "start": 5,
          "trailingComments": undefined,
          "type": "Identifier",
        },
        "start": 1,
        "trailingComments": undefined,
        "type": "MemberExpression",
      },
      "end": 13,
      "innerComments": undefined,
      "leadingComments": undefined,
      "loc": SourceLocation {
        "end": Position {
          "column": 13,
          "line": 1,
        },
        "start": Position {
          "column": 1,
          "line": 1,
        },
      },
      "start": 1,
      "trailingComments": undefined,
      "type": "CallExpression",
    },
    "computed": false,
    "path": Node {
      "end": 8,
      "innerComments": undefined,
      "leadingComments": undefined,
      "loc": SourceLocation {
        "end": Position {
          "column": 8,
          "line": 1,
        },
        "identifierName": "bar",
        "start": Position {
          "column": 5,
          "line": 1,
        },
      },
      "name": "bar",
      "start": 5,
      "trailingComments": undefined,
      "type": "Identifier",
    },
  },
  Object {
    "argumentsPath": null,
    "computed": true,
    "path": Node {
      "end": 22,
      "innerComments": undefined,
      "leadingComments": undefined,
      "loc": SourceLocation {
        "end": Position {
          "column": 22,
          "line": 1,
        },
        "identifierName": "baz",
        "start": Position {
          "column": 19,
          "line": 1,
        },
      },
      "name": "baz",
      "start": 19,
      "trailingComments": undefined,
      "type": "Identifier",
    },
  },
  Object {
    "argumentsPath": null,
    "computed": true,
    "path": Node {
      "end": 26,
      "loc": SourceLocation {
        "end": Position {
          "column": 26,
          "line": 1,
        },
        "start": Position {
          "column": 24,
          "line": 1,
        },
      },
      "raw": "42",
      "start": 24,
      "type": "Literal",
      "value": 42,
    },
  },
]
`;
